### PR TITLE
Use Github actions as test workflow

### PR DESCRIPTION
**Pull request by clemsos**
# Description

relates to #7658

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by clemsos** [comment]
ok getting there... the monorepo check and the tests are running now.

**Comment by julien51** [comment]
Nice!

File: scripts/monorepo.sh [hunk 1]
```diff
@@ -12,7 +12,8 @@ if [ ! "$BRANCH" = "master" ]; then
   HAS_CHANGES=$(git diff `git merge-base HEAD origin/master` --name-only -- $SHARED_CONFIG_FILES $SERVICE)
   if [ ! -n "$HAS_CHANGES" ]; then
     echo "No change in $SERVICE, skipping job."
-    circleci step halt
+  else
+    echo "changed"
```
Comment (julien51) [inline scripts/monorepo.sh:16]: Can we change the circle config to just call `circleci step halt` if result is `changed` ?

**Comment by julien51** [comment]
Looks good so far! I think we're missing a few other tasks, like deployments?

File: smart-contracts/README.md [hunk 2]
```diff
@@ -188,7 +188,7 @@ Deploying Governor on localhost with the account: 0xf39Fd6e51aad88F6F4ce6aB88272
 }
 ```
 
-check [`./proposals/000-example.js`](./proposals/000-example.js) for an example.
+check [`./proposals/000-example.js`](./proposals/000-example.js) for an example
```
Comment (julien51) [inline smart-contracts/README.md:191]: oh, I see why you did this :)

**Comment by clemsos** [comment]
> Looks good so far! I think we're missing a few other tasks, like deployments?

yep, on its way :)

File: scripts/monorepo.sh (hunk 1 above)
Comment (clemsos) [reply scripts/monorepo.sh:16]: yep I added to circleci conf directly

**Comment by clemsos** [comment]
ok seems like we are all setup. 

Developing for CI is a complete pain as there is no way to test things locally - therefore such messy commits... Now we still have to test how things work when merged on master.

One remaining problem is that we currently have both circle & gh actions setup. For the tests is not much of a problem, but not sure what it will look like when deployed.

Also for deployments, we still need a cleaner way to populate env vars properly. Gh actions is better in this sense - we can pass a `.env` directly to `gh` cli with an environment assigned.

Anyways not really sure whats the best way fwd. I will say just merge and see, but we may have unexpected things happening...

**Comment by julien51** [comment]
Yes, agreed. merging is probably the only way to find bugs issue, so I'd say.... YOLO?

File: .circleci/config.yml [hunk 3]
```diff
@@ -109,14 +117,6 @@ jobs:
           command: |
             sh ./scripts/heroku.sh << parameters.service >> << parameters.app-name >>
 
-  open-prod-pull-request:
```
Comment (julien51) [inline .circleci/config.yml:112]: Hum. Can we keep it for now? I think it's ok that it opens 2 prod deploys for now (we can still close one!)

File: .circleci/config.yml [hunk 4]
```diff
@@ -201,15 +201,4 @@ workflows:
             branches:
               only: master
           requires:
-            - integration-tests
-
-  prepare-prod-deploy:
```
Comment (julien51) [inline .circleci/config.yml:206]: same!

File: .github/actions/docker-cache/backup.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
```
Comment (julien51) [inline .github/actions/docker-cache/backup.sh:2]: Ohhh, do you mind adding a few comments at the top about the purpose of this?

File: .github/actions/docker-cache/restore.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
```
Comment (julien51) [inline .github/actions/docker-cache/restore.sh:2]: same?

File: .github/workflows/_heroku.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+name: Deploy on Heroku
```
Comment (julien51) [inline .github/workflows/_heroku.yml:1]: Ha! Mind telling me more about the change here?

**Comment by julien51** [comment]
also I see tests did not run for this PR because it did not detect any change... which kind of makes sense but also I think it would be good to have a way to "force" all tests to run in a given PR. Maybe by changing the name of the PR and adding  "force-tests" in the title?

**Comment by clemsos** [comment]
> it would be good to have a way to "force" all tests to run in a given PR. Maybe by changing the name of the PR and adding "force-tests" in the title?

ok added. The way to run all test is to add the label "run-all-tests" to the PR. I thought it was cleaner than polluting the PR title, as it gets added un master when PR is squashed.

File: .github/workflows/deploy.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+on: 
+  push:
+    branches:
+      - master
+      - production
+
+jobs:
+  # heroku deploy 
+  deploy-locksmith-staging:    
+    if: contains(github.base_ref, 'master')
```
Comment (julien51) [inline .github/workflows/deploy.yml:11]: Let's disable for now until we get the secrets figured out!

**Review by julien51 — CHANGES_REQUESTED** [review]
Please disable deployments and then LG!

**Comment by clemsos** [comment]
ok moved to a `disabled` folder

**Comment by clemsos** [comment]
```
Summary of all failing tests
FAIL __tests__/controllers/metadataController/tokenData.test.ts
  ● Requesting Token Data › when persisted data doesn't exist › returns wellformed data for Week in Ethereum News

    expect(received).toEqual(expected) // deep equality

    Expected: ObjectContaining {"description": "A Key to the 'Week in Ethereum News' lock. Unlock is a protocol for memberships. https://unlock-protocol.com/", "image": "https://assets.unlock-protocol.com/nft-images/week-in-ethereum.png", "name": "Unlock Key to Week in Ethereum News"}
    Received: {"address": "0x95de5F777A3e283bFf0c47374998E10D8A2183C7", "attributes": [{"display_type": "number", "trait_type": "expiration", "value": 1567190711}], "description": "we are chilling and such", "expiration": 1567190711, "image": "http://image.location.url", "name": "An awesome Lock", "owner": "0xabcd"}

      149 |
      150 |       expect(response.status).toBe(200)
    > 151 |       expect(response.body).toEqual(
          |                             ^
      152 |         expect.objectContaining({
      153 |           description:
      154 |             "A Key to the 'Week in Ethereum News' lock. Unlock is a protocol for memberships. https://unlock-protocol.com/",

      at Object.<anonymous> (__tests__/controllers/metadataController/tokenData.test.ts:151:29)

  ● Requesting Token Data › when the persisted data exists › returns key specific information when available

    expect(received).toEqual(expected) // deep equality

    Expected: ObjectContaining {"custom_item": "custom value", "description": "A Key to the 'Week in Ethereum News' lock. Unlock is a protocol for memberships. https://unlock-protocol.com/", "image": "https://assets.unlock-protocol.com/nft-images/week-in-ethereum.png", "name": "Unlock Key to Week in Ethereum News"}
    Received: {"address": "0x95de5F777A3e283bFf0c47374998E10D8A2183C7", "attributes": [{"display_type": "number", "trait_type": "expiration", "value": 1567190711}], "custom_item": "custom value", "description": "we are chilling and such", "expiration": 1567190711, "image": "http://image.location.url", "name": "An awesome Lock", "owner": "0xabcd"}

      190 |
      191 |       expect(response.status).toBe(200)
    > 192 |       expect(response.body).toEqual(
          |                             ^
      193 |         expect.objectContaining({
      194 |           description:
      195 |             "A Key to the 'Week in Ethereum News' lock. Unlock is a protocol for memberships. https://unlock-protocol.com/",

      at Object.<anonymous> (__tests__/controllers/metadataController/tokenData.test.ts:192:29)
```


This is likely failing because of env vars not properly set?

**Comment by julien51** [comment]
Possibly? I can look it up later today!

**Comment by julien51** [comment]
Not sure this is an Env var issue because it passes locally and I don't have special env vars :/

**Comment by julien51** [comment]
It actually looks like some data in the database does not get reset between tests!

**Pull request merged**